controlplane/kubeadm/internal/machine_collection.go

```diff
@@ -134,3 +135,13 @@ func (s FilterableMachineCollection) DeepCopy() FilterableMachineCollection {
 	}
 	return result
 }
+
+// ConditionGetters returns the slice with machines converted into conditions.Getter.
+func (s FilterableMachineCollection) ConditionGetters() []conditions.Getter {
+	res := make([]conditions.Getter, 0, len(s))
+	for _, v := range s {
+		value := *v
+		res = append(res, &value)
+	}
+	return res
+}
```

Comment:
Once the conditions PR for Machines is merged, it looks like you could replace the call here by making `unsortedList` public and calling that instead.

Reply:
Unfortunately, this does not work: 

> cannot use ownedMachines.UnsortedList() (type []*"sigs.k8s.io/cluster-api/api/v1alpha3".Machine) as type []conditions.Getter in argument to conditions.SetAggregateCondition